### PR TITLE
Transform numpy vars to tensors

### PR DESCRIPTION
**Pull request by im-a-jin**
Reordering the if statements allows `torch.ToTensor` to be applied without breaking things. `mask_array` is cast to `np.int64` as its default type, `np.uint64`, isn't supported by some pytorch functions.

This method still results in errors in some cases. See followup comment for my thoughts on implementation issues.

### CODE REVIEW

**Comment by im-a-jin** [comment]
The problem I'm encountering is that the transpose statements are based on the assumption that a `torch.ToTensor` transform is passed to the dataset. If it isn't a different order of axes transposes are needed. I'm not sure what the best way to address this would be.

Personally, I would prefer if ToTensor is just not used, but that's putting expectations on the users which may not always be satisfied.

**Comment by im-a-jin** [comment]
Ok, I think this is my final update. Of note:

Images are returned in channel first mode (C, X, Y) for 2D images and (C, Z, X, Y) for 3D images. The 3D shape is consistent with the result of the old ver with the torch.ToTensor transform. The 2D shape was originally (C, Y, X) after torch.ToTensor which would not have impacted training (since everything's just flipped), but I swapped the last two axes for consistency

I moved the transform if statements into the 2D block as pytorch does not support transforms for 3D images, and the ToTensor scaling ([0, 255] --> [0, 1]) is integrated into the dataset code. I'm not sure if this scaling step is desirable as it modifies the data in a nontrivial manner.

**Comment by im-a-jin** [comment]
I'm getting outputs where image_arrays of format (z, y, x) map to mask_arrays of (x, y). Can someone verify?

**Comment by erikjohnson24** [comment]
Some minor notes! For task 2, we need to change the defaults to --task MTNeuro/taskconfig/task2_2D_3class.json --network MTNeuro/networkconfig/UNet_2D_3class.json 
Installing on python3.8 I found cloud-volume and pyrsistent were missing dependencies. will continue verifying new functionality with torch tensors before merging.

**Comment by im-a-jin** [comment]
Looks like task 1 is also reliant on the toTensor transform being applied to the mask array which stores a 256x256 "image" of roi labels for each example. Is there a reason for this pixel-level region labeling? The BossDBDataset class takes the mode from the flattened mask_array which requires `torch.mode`.

**Comment by erikjohnson24** [comment]
ToTensor addressed in broader PR #27 closing
